lib/knapsack_pro/client/connection.rb

```diff
@@ -141,13 +141,24 @@ def make_request(&block)
       end
 
       def build_http(uri)
-        http = Net::HTTP.new(uri.host, uri.port)
+        http = net_http.new(uri.host, uri.port)
         http.use_ssl = (uri.scheme == 'https')
         http.open_timeout = TIMEOUT
         http.read_timeout = TIMEOUT
         http
       end
 
+      def net_http
+        if defined?(WebMock) &&
+            defined?(WebMock::HttpLibAdapters) &&
+            defined?(WebMock::HttpLibAdapters::NetHttpAdapter) &&
+            defined?(WebMock::HttpLibAdapters::NetHttpAdapter::OriginalNetHTTP)
```

Comment:
Is there a reason why we need so many `defined?`?

I think `defined?(WebMock::HttpLibAdapters::NetHttpAdapter::OriginalNetHTTP)` is enough because Ruby will resolve the entire chain via :: only if the modules and classes are defined.

For example I can do this in the irb and it won't blow up.

```
3.2.3 :004 > defined?(SomethingFake::Fake2::Fake3)
 => nil
```

Reply:
You are correct. Updated 🙂

I tested it on both ruby 3.3 and 2.7.